engine/flink/tests/src/test/scala/pl/touk/nussknacker/devmodel/MultiSelectEditorTest.scala

```diff
@@ -0,0 +1,188 @@
+        "multiSelectParam" -> jsonExpressionString.jsonExpression
+      )
+      .emptySink("end", TestScenarioRunner.testResultSink, "value" -> "#serviceOut".spel)
+    testScenarioRunner.runWithData[Int, AnyRef](scenario, List(1))
+  }
+
+}
+
+object MultiSelectDynamicParametersBasedComponent extends EagerService with SingleInputDynamicComponent {
+
+  private val multiSelectParameterName = ParameterName("multiSelectParam")
+
+  import scala.jdk.CollectionConverters._
+
+  private val multiSelectParameter = Parameter
+    .optional[Any](multiSelectParameterName)
+    .copy(
+      editors = List(
+        MultiSelectEditor(
+          List(
+            MultiSelectFixedValue(Json.fromString("stringOption"), "stringOption"),
+            MultiSelectFixedValue(Json.fromInt(3), "intOption"),
+            MultiSelectFixedValue(Json.arr(Json.fromInt(1), Json.fromInt(2), Json.fromInt(3)), "listOption"),
+            MultiSelectFixedValue(Json.obj("key" -> Json.fromString("value")), "recordOption")
+          )
+        )
+      )
+    )
+
+  override type Implementation = ServiceInvoker
+  override type State          = Unit
+
+  override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])(
+      implicit nodeId: NodeId
+  ): MultiSelectDynamicParametersBasedComponent.ContextTransformationDefinition = {
+    case TransformationStep(Nil, _) => NextParameters(List(multiSelectParameter))
+    case TransformationStep((multiSelectParameterName, DefinedEagerParameter(_, returnType)) :: Nil, _) =>
+      val outName = OutputVariableNameDependency.extract(dependencies)
+      FinalResults(context.withVariableUnsafe(outName, returnType), List.empty)
+  }
+
+  override def implementation(
+      params: Params,
+      dependencies: List[NodeDependencyValue],
+      finalState: Option[Unit]
+  ): MultiSelectDynamicParametersBasedComponent.Implementation = {
+    val paramValue = params.extractDeclaredParamUnsafe[Json](multiSelectParameterName)
+    new ServiceInvoker {
+      override def invoke(context: Context)(
+          implicit ec: ExecutionContext,
+          collector: InvocationCollectors.ServiceInvocationCollector,
+          componentUseContext: ComponentUseContext
+      ): Future[Any] = Future(paramValue)
+    }
+  }
+
+  override def nodeDependencies: List[NodeDependency] = List.empty
+}
+
+object MultiSelectAnnotationBasedComponent extends Service with Serializable {
+
+  @MethodToInvoke
+  def invoke(
+      @ParamName("multiSelectParam")
+      @Editor(
+        `type` = EditorType.MULTI_SELECT_EDITOR,
+        // Only String values may be configured using annotation based component API
+        possibleMultiSelectValues = Array(
+          new MultiSelectLabeledValue(value = "option1", label = "option1"),
+          new MultiSelectLabeledValue(value = "option2", label = "option2")
+        )
+      )
+      @Editor(`type` = EditorType.JSON_EDITOR)
+      multiSelect: Any,
```

Comment:
same like here -> https://github.com/TouK/nussknacker/pull/8714/files#r2554971616

Reply:
changed it :+1: